<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>InteractiveSegmentTubes</class>
 <widget class="qSlicerWidget" name="InteractiveSegmentTubes">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>299</width>
    <height>362</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Register</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="3" column="0" colspan="2">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="5" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="ApplyPushButton">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start running the &lt;span style=&quot; font-style:italic;&quot;&gt;Segment Tubes&lt;/span&gt; CLI to process all the &lt;span style=&quot; font-weight:600;&quot;&gt;Seed points&lt;/span&gt;.&lt;/p&gt;&lt;p&gt;The module will be run any time there is a &lt;span style=&quot; font-weight:600;&quot;&gt;'Queued'&lt;/span&gt; seed in &lt;span style=&quot; font-weight:600;&quot;&gt;Seed points&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>Start</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="ctkCollapsibleButton" name="CollapsibleButton">
     <property name="text">
      <string>IO</string>
     </property>
     <property name="contentsFrameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="horizontalSpacing">
       <number>9</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="InputLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string/>
        </property>
        <property name="text">
         <string>Input image</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="InputNodeComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Choose the image the vessels will be extracted from</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="baseName">
         <string/>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="editEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="OutputLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Choose the spatial object the vessels will be extracted to.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Note that the new vessels will be automatically &lt;span style=&quot; font-weight:600;&quot;&gt;merged to &lt;/span&gt;the current &lt;span style=&quot; font-weight:600;&quot;&gt;output tube file&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Output tube </string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="OutputNodeComboBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Choose the spatial object the vessels will be extracted to.&lt;/p&gt;&lt;p&gt;Note that the new vessels will be automatically &lt;span style=&quot; font-weight:600;&quot;&gt;merged to &lt;/span&gt;the current &lt;span style=&quot; font-weight:600;&quot;&gt;output tube file&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLSpatialObjectsNode</string>
         </stringlist>
        </property>
        <property name="showHidden">
         <bool>true</bool>
        </property>
        <property name="baseName">
         <string>Vessels</string>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="qSlicerCLIProgressBar" name="CLIProgressBar"/>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="ctkCollapsibleButton" name="CollapsibleButton_2">
     <property name="text">
      <string>Seeds</string>
     </property>
     <property name="contentsFrameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="4" column="0">
       <widget class="QLabel" name="SeedsShowStatusLabel">
        <property name="text">
         <string>Show seed status</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="qMRMLNodeComboBox" name="SeedPointNodeComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select here the list of seeds to initiate the&lt;span style=&quot; font-style:italic;&quot;&gt; Segment Tubes&lt;/span&gt; CLI with.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLMarkupsFiducialNode</string>
         </stringlist>
        </property>
        <property name="showHidden">
         <bool>false</bool>
        </property>
        <property name="baseName">
         <string>Seeds</string>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="editEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="SeedPointLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Seed points</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="SeedsSizeLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Seeds displayed size</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1" colspan="2">
       <widget class="QCheckBox" name="SeedsShowStatusCheckBox">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Show the seeds status or not.&lt;/p&gt;&lt;p&gt;A seed can be &lt;span style=&quot; font-weight:600;&quot;&gt;'Queued'&lt;/span&gt; (waiting to be processed) or &lt;span style=&quot; font-weight:600;&quot;&gt;'Processed&lt;/span&gt;' (after it went through the &lt;span style=&quot; font-style:italic;&quot;&gt;Segment Tubes&lt;/span&gt; CLI).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="qMRMLSliderWidget" name="SeedScaleSliderWidget">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Controls the &lt;span style=&quot; font-weight:600;&quot;&gt;scale&lt;/span&gt; of the vessels in the &lt;span style=&quot; font-style:italic;&quot;&gt;Segment Tubes&lt;/span&gt; CLI.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;By default, this is linked to the displayed &lt;span style=&quot; font-weight:600;&quot;&gt;Seed size&lt;/span&gt;. The two parameters can be controled separatly by clicking the button with the chain icon. &lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Note that the &lt;span style=&quot; font-weight:600;&quot;&gt;sale&lt;/span&gt; used in the CLI is the &lt;span style=&quot; font-weight:600;&quot;&gt;scale&lt;/span&gt; at the moment the CLI is ran (and *NOT* when the &lt;span style=&quot; font-weight:600;&quot;&gt;seed&lt;/span&gt; is clicked).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="pageStep">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>20.000000000000000</double>
        </property>
        <property name="value">
         <double>3.000000000000000</double>
        </property>
        <property name="quantity">
         <string>Length</string>
        </property>
        <property name="unitAwareProperties">
         <set>qMRMLSliderWidget::Precision|qMRMLSliderWidget::Prefix|qMRMLSliderWidget::Suffix</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLSliderWidget" name="SeedsSizeSliderWidget">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Control the seeds' point size.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="pageStep">
         <double>0.100000000000000</double>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>20.000000000000000</double>
        </property>
        <property name="value">
         <double>3.000000000000000</double>
        </property>
        <property name="quantity">
         <string>Length</string>
        </property>
        <property name="unitAwareProperties">
         <set>qMRMLSliderWidget::Precision|qMRMLSliderWidget::Prefix|qMRMLSliderWidget::Suffix</set>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="SeedScaleLabel">
        <property name="text">
         <string>Seed scale</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2" rowspan="2">
       <widget class="QPushButton" name="SizeAndScaleLinkedButton">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Link&lt;/span&gt; or &lt;span style=&quot; font-weight:600;&quot;&gt;Unlink&lt;/span&gt; the &lt;span style=&quot; font-weight:600;&quot;&gt;seed scale&lt;/span&gt; and the &lt;span style=&quot; font-weight:600;&quot;&gt;seed displayed size&lt;/span&gt;.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;When linking back the two together, the &lt;span style=&quot; font-weight:600;&quot;&gt;seed scale&lt;/span&gt; will snap to the &lt;span style=&quot; font-weight:600;&quot;&gt;seed displayed size.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="2">
       <widget class="QComboBox" name="ParameterFileComboBox">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select a &lt;span style=&quot; font-weight:600;&quot;&gt;Parameter file&lt;/span&gt; from the list of available file. It will be used to configure the &lt;span style=&quot; font-style:italic;&quot;&gt;Segment Tubes&lt;/span&gt; CLI each time it's ran.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;It is possible to select your own file using the &lt;span style=&quot; font-weight:600;&quot;&gt;Load from file...&lt;/span&gt; option. A dialog be prompt to choose what file to use. This new file will be added to the list of default files.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="insertPolicy">
         <enum>QComboBox::InsertAtTop</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Parameter file</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLSliderWidget</class>
   <extends>ctkSliderWidget</extends>
   <header>qMRMLSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qSlicerCLIProgressBar</class>
   <extends>QWidget</extends>
   <header>qSlicerCLIProgressBar.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>InteractiveSegmentTubes</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>InputNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>222</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>195</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>InteractiveSegmentTubes</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>OutputNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>408</x>
     <y>121</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>InteractiveSegmentTubes</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>SeedPointNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>189</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>84</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>InteractiveSegmentTubes</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>SeedsSizeSliderWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>303</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>332</x>
     <y>142</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
